### PR TITLE
BF: workaround for exec not being parsed (?) correctly on python 2.7.3

### PR DESCRIPTION
**Pull request by yarikoptic**

### CODE REVIEW

**Comment by yarikoptic** [comment]
this one seems to work on both precise and with python3

File: mdp/utils/templet.py [hunk 1]
```diff
@@ -128,6 +128,14 @@ class RecipeWithPriceTemplate(RecipeTemplate):
...
 from future.utils import with_metaclass
 
+# There is an issue with exec(st, g, l) not being treated correctly with python
+# 2.7.3 on precise/travis.  Since it spits SyntaxError, decision between the two
+# needs to be done at code parsing stage thus here we define the adapter function
+if sys.version_info[:3] < (2, 7, 9):
+    from ._exec import _exec
+else:
+    def _exec(st, g, l):
+        exec(st, g, l)
```
Comment (yarikoptic) [inline mdp/utils/templet.py]: FWIW -- can't be just `_exec = exec` since then python 2.7.x gets puky

**Comment by yarikoptic** [comment]
FWIW, with even previous non-python3 friendly fix our tests for PyMVPA started to pass without puking on mdp imports on travis:  https://travis-ci.org/PyMVPA/PyMVPA/builds/114254331

**Comment by keszybz** [comment]
Something strange is going on. I tested the exec line on a Precise VM and in python2.7.3 I compiled from git, and it works fine in both. Do you maybe have a link to the failing test case (with the code before your patch)?

**Comment by yarikoptic** [comment]
it could well be some patch within ubuntu precise?  I am just testing in precise chroot:

```
$> schroot -c precise                                                            
W: Shell ‘/bin/zsh’ not available: /bin/zsh: Failed to stat file: No such file or directory
W: Falling back to shell ‘/bin/bash’
I: New layout of history files also found.
   Please remove/propagate old history into new
(git)~/deb/gits/mdp:[master]
#1 !1087 [0].....................................:Mon Mar 07 16:53:56:.
hopa(precise-amd64):~/deb/gits/mdp
$> source venv-precise/bin/activate
(venv-precise)(git)~/deb/gits/mdp:[master]
#2 !1088 [0].....................................:Mon Mar 07 16:53:59:.
hopa(precise-amd64):~/deb/gits/mdp
$> pip install -e .
Obtaining file:///home/yoh/deb/gits/mdp
  Running setup.py egg_info for package from file:///home/yoh/deb/gits/mdp
    
Requirement already satisfied (use --upgrade to upgrade): numpy in /usr/lib/python2.7/dist-packages (from MDP==3.4)
Requirement already satisfied (use --upgrade to upgrade): future in ./venv-precise/lib/python2.7/site-packages (from MDP==3.4)
Installing collected packages: MDP
  Running setup.py develop for MDP
    
    Creating /home/yoh/deb/gits/mdp/venv-precise/lib/python2.7/site-packages/MDP.egg-link (link to .)
    MDP 3.4 is already the active version in easy-install.pth
    
    Installed /home/yoh/deb/gits/mdp
Successfully installed MDP
Cleaning up...

$> python -c 'import mdp'
/usr/local/lib/python2.7/dist-packages/joblib/_multiprocessing_helpers.py:29: UserWarning: [Errno 13] Permission denied.  joblib will operate in serial mode
  warnings.warn('%s.  joblib will operate in serial mode' % (e,))
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "mdp/__init__.py", line 131, in <module>
    from . import utils
  File "mdp/utils/__init__.py", line 29, in <module>
    from .slideshow import (basic_css, slideshow_css, HTMLSlideShow,
  File "mdp/utils/slideshow.py", line 23, in <module>
    from . import templet
  File "mdp/utils/templet.py", line 196
    exec(code, globals, kw)
SyntaxError: unqualified exec is not allowed in function 'expand' it is a nested function
```

**Comment by keszybz** [comment]
Thanks for the links. I just didn't test nested functions.

It should be enough to define a simple wrapper function. Patch coming right up.

**Comment by keszybz** [comment]
Could you check if this patch works for you?

**Comment by keszybz** [comment]
Closing in favour of #17.

**Pull request closed without merging**